### DISCUSSION TITLE
Checking stocklocations

### DISCUSSION THREAD

**Discussion by diodenschein** [inventree/InvenTree#4682] [Ideas]
I would quite like to be able to check stocklocations and have an action logged in the stock history for tracked stockitems too. 

My usecase is the following: 
I have a box full of stock and can do a stock take for untracked items. But I would like to be able to also scan all the serial numbers in a storage location into it again and update the stockitems' date to confirm it's still there. 

I guess I would then go searching for the ones that haven't been updated in that stocktake.

Is there a reason "scan into location" prevents this?

We essentially run an electronics manufacturing business that also hires out our assembled stock in various configurations. I am currently trying to work out how to manage rented out stock with stocklocations that I move from our storage to an event and back. (Which I think works great)
But I occasionally would like to confirm boxes/ pallets still contain whatever they should.

**Comment by matmair** [comment]
Sounds like a specific workflow, a [plugin](https://docs.inventree.org/en/stable/extend/plugins/) would probably be the best solution for that.

**Comment by SchrodingersGat** [comment]
@diodenschein are you looking for the ability to "check items in" to a location they are already located in? Am I reading this right?

**Reply by diodenschein** [reply]
Yes. Basically logging it's been "seen" in the stock tracking instead of complaining it's already in there.

This is so you know it's been "seen" in that stocklocation and have a record of it.

**Reply by matmair** [reply]
Not really sure if this is a common request; do you know about the stock count function?

**Reply by wolflu05** [reply]
Sounds like inventory? (German: Inventur)

**Reply by diodenschein** [reply]
I do know about the stock count function. 

Ah yes I'm dealing with individually tracked stock items with serial numbers and barcodes. Count doesn't really work on them.

So if someone takes one out and doesn't track it. Which does happen. I'd like a method of scanning all stockitems that are still in the location to find out which ones are still there. 


Does that make more sense?

**Reply by matmair** [reply]
I do not think this is a common usage pattern, first that I hear of this. Thoughts @inventree/triage?

**Reply by wolflu05** [reply]
Maybe this can be implemented via a plugin?

**Reply by SchrodingersGat** [reply]
@diodenschein I think I understand what you are saying. Currently if you try to scan an item "into" a location, it will reject the action, as the item is already there. But what you want is to update the "last updated" date on the stockitem, based on the user who scanned.

I think this actually makes sense.

Are you using the web interface for barcode scanning or the app?

**Reply by diodenschein** [reply]
@SchrodingersGat  Yes something like that. I think that's better  behavior and useful. 

We do use both. 
I would be more than happy if it worked in the web interface. I think it makes more sense there especially when using a handheld scanner that is a lot faster.

**Reply by SchrodingersGat** [reply]
@diodenschein see https://github.com/inventree/InvenTree/pull/4757
